### PR TITLE
feat: support jest v26

### PR DESCRIPTION
**Pull request by Moeriki**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,15 +40,15 @@
     "eslint-plugin-jest": "^22.6.4",
     "eslint-plugin-prettier": "^3.1.0",
     "execa": "^1.0.0",
-    "jest": "^21.0.0 || ^22.0.0 || ^23.0.0 || ^24.0.0 || ^25.1.0",
+    "jest": "^26.0.0",
```
Comment (ljharb) [inline package.json:43]: This line needs to exactly match the peerdep.

Comment (Moeriki) [reply package.json:43]: Ah sorry thought it was an oversight. Doesn't npm just install the latest?

**Comment by Moeriki** [comment]
It seems the snapshot `normalize` isn't effective anymore. I get snapshot failures because of test duration. I'm still checking why.

File: package.json (hunk 1 above)
Comment (ljharb) [reply package.json:43]: If none is otherwise installed, yes; but in tests, we'll want to install each supported version, and `npm ls` should pass there.

**Comment by Moeriki** [comment]
Had to update

```diff
- /\(?\d*\.?\d+m?s\)?/g
+ /\(?\d*\.?\d+ m?s\b\)?/g
```

Space because jest updated its duration formating, and `\b` because otherwise it would match with strings like `1 skipped`.

File: integrationTests/__snapshots__/custom-parser.test.js.snap [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Doesn't override parser when not set 1`] = `
+exports[`Doesn't override parser when not set [ESLint<6] 1`] = `
```
Comment (SimenB) [inline integrationTests/__snapshots__/custom-parser.test.js.snap:3]: seems like this is failing on CI?

File: .travis.yml [hunk 3]
```diff
@@ -35,3 +38,10 @@ matrix:
       env: 'ESLINT=5 JEST=25'
     - node_js: '8'
```
Comment (SimenB) [inline .travis.yml:39]: we can run jest 25 on node 8, the bug was fixed. not really for this PR though 🙂

File: .travis.yml [hunk 4]
```diff
@@ -35,3 +38,10 @@ matrix:
...
       env: 'ESLINT=6 JEST=25'
+    # exclude jest 26 on node 8
```
Comment (SimenB) [inline .travis.yml:41]: need to skip it on node 6 as well

File: integrationTests/__snapshots__/custom-parser.test.js.snap (hunk 2 above)
Comment (Moeriki) [reply integrationTests/__snapshots__/custom-parser.test.js.snap:3]: The snapshot was locally updated by jest 26. It seems they strip it for a reason. I'm checking the release notes.

File: .travis.yml (hunk 3 above)
Comment (Moeriki) [reply .travis.yml:39]: Ah I see. I was going to ask :)

File: .travis.yml (hunk 4 above)
Comment (Moeriki) [reply .travis.yml:48]: Fixed.

**Comment by Moeriki** [comment]
I think I misunderstood something in the testing setup.

Had some failures and updated some snapshots, while I shouldn't have.

All seems good now.

File: integrationTests/__snapshots__/custom-parser.test.js.snap (hunk 2 above)
Comment (Moeriki) [reply integrationTests/__snapshots__/custom-parser.test.js.snap]: Nevermind I did something wrong. They all pass!

**Comment by SimenB** [comment]
`v0.8.0` published, thanks @Moeriki!

**Pull request merged**